cost-analyzer/templates/cost-analyzer-deployment-template.yaml

```diff
@@ -713,10 +713,50 @@ spec:
             - name: GOOGLE_ANALYTICS_TAG
               value: {{ .Values.reporting.googleAnalyticsTag }}
             {{- end }}
+            {{- /*
+              Leader/Follower has baseline requirements before enabling:
+                * ETL FileStore Enabled 
+                * Bucket Backup Configured
+                * Replicas > 1
+            */}}
+            {{- if .Values.kubecostDeployment }}
+            {{- if .Values.kubecostDeployment.leaderFollower }}
+            {{- if .Values.kubecostDeployment.leaderFollower.enabled -}}
+
+            {{- $etlFileStore := .Values.kubecostModel.etlFileStoreEnabled }}
+            {{- if not $etlFileStore }}
+            {{- fail "Leader/Follower requires kubecostModel.etlFileStoreEnabled be true." }}
+            {{- end -}}
+            
+            {{- if (eq (quote .Values.kubecostModel.etlBucketConfigSecret) "") }}
+            {{- fail "Leader/Follower requires kubecostModel.etlBucketConfigSecret be valid." }}
+            {{- end -}}
+
+            {{- $replicas := .Values.kubecostDeployment.replicas | default 1 }}
+            {{- if not (gt (int $replicas) 1) }}
+            {{- fail "Leader/Follower should be used with kubecostDeployment.replicas > 1" }}
+            {{- end }}
+            {{- /* 
+              Checks to ensure that the named template returns true if we've made it here 
+            */}}
+            {{- $result := include "cost-analyzer.leaderFollowerEnabled" . }}
+            {{- if not $result }}
+            {{- fail (quote $result) }}
+            {{- end }}
+            - name: LEADER_FOLLOWER_ENABLED
+              value: "true"
+            {{- end }}
+            {{- end }}
+            {{- end }}
```

Comment:
This seems redundant. Can we just add the failures to the defined template `cost-analyzer.leaderFollowerEnabled` and then just check the first value for "true" otherwise re-fail the error up?

(I am still a n00b at templating so sorry if my understanding it wrong.)

Reply:
We could go that route, but we would lose the details regarding what caused the failure. The only reason the `"cost-analyzer.leaderFollowerEnabled"` is useful is as an all-in-one check, but since we don't control ordering for how the templates are generated, we need at least one place that hard-fails with good error messaging, and the other checks can just use the template.

Reply:
You may be correct about the template -- just have the hard failures (rather than return a true/false value). And you're right about the redundancy here with respect to calling the template in the immediate checks... It's redundant granted that both the template and my check in the cost-analyzer-deployment are logically equivalent (which I was sort of using as a unit test). We could technically remove the template call, but it's not necessarily hurting anything by leaving it (in fact, it helps me sleep at night).

Reply:
ah, so `cost-analyzer.leaderFollowerEnabled` gets computed once? I thought we could just do something like the following to "re-raise".

```
{{- $result, $err := include "cost-analyzer.leaderFollowerEnabled" . }}
{{- if $err }}
{{- fail (quote $err) }}
```

Again, sorry for the n00b-ness on this, but it sounds like you have thought it through. Go forth!

Reply:
> (I am still a n00b at templating so sorry if my understanding it wrong.)
The helm templating is truly the most counter-intuitive tech I have ever used. However, I think you're understanding correctly -- In general though, I have found that using the long-winded approach (which includes some possible redundancy) is good way to ensure consistent output across "random" template ordering. The biggest issue IMO is the yaml types (especially bool) converting into strings where `"false"` is `true` and `""` is `false`.